File: arrays/2799. Count Complete Subarrays in an Array/countCompleteSubarrays.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countCompleteSubarrays = function(nums) {
    let count = 0;
    let totalDistinct = new Set(nums).size;

    for (let i = 0; i < nums.length; i++) {
        let temp = new Set();
        for (let j = i; j < nums.length; j++) {
            temp.add(nums[j]);

            if (temp.size === totalDistinct) count++;
        }
    }

    return count;
};

File: arrays/2799. Count Complete Subarrays in an Array/countCompleteSubarrays.py
class Solution(object):
    def countCompleteSubarrays(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        total_distinct = len(set(nums))
        count = 0

        for i in range(len(nums)):
            temp = set()
            for j in range(i, len(nums)):
                temp.add(nums[j])
                if len(temp) == total_distinct:
                    count += 1
        
        return count